src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PreparedStatementTest.java

```diff
@@ -0,0 +1,249 @@
+
+        try (SQLServerConnection con = (SQLServerConnection)DriverManager.getConnection(connectionString)) {
+            conOuter = con;
+
+            // Clean-up proc cache
+            this.executeSQL(con, "DBCC FREEPROCCACHE;"); 
+            
+            String lookupUniqueifier = UUID.randomUUID().toString();
+
+            String queryCacheLookup = String.format("/*unpreparetest_%s%%*/SELECT * FROM sys.tables;", lookupUniqueifier);
+            String query = String.format("/*unpreparetest_%s only sp_executesql*/SELECT * FROM sys.tables;", lookupUniqueifier);
+
+            // Verify nothing in cache.
+            String verifyTotalCacheUsesQuery = String.format("SELECT CAST(ISNULL(SUM(usecounts), 0) AS INT) FROM sys.dm_exec_cached_plans AS p CROSS APPLY sys.dm_exec_sql_text(p.plan_handle) AS s WHERE s.text LIKE '%%%s'", queryCacheLookup);
+
+            assertSame(0, executeSQLReturnFirstInt(con, verifyTotalCacheUsesQuery));
+
+            int iterations = 25;
+
+            // Verify no prepares for 1 time only uses.
+            for(int i = 0; i < iterations; ++i) {
+                try (SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement)con.prepareStatement(query)) {
+                    pstmt.execute();
+                } 
+                assertSame(0, con.getDiscardedServerPreparedStatementCount());
+            }
+
+            // Verify total cache use.
+            assertSame(iterations, executeSQLReturnFirstInt(con, verifyTotalCacheUsesQuery));
+
+            query = String.format("/*unpreparetest_%s, sp_executesql->sp_prepexec->sp_execute- batched sp_unprepare*/SELECT * FROM sys.tables;", lookupUniqueifier);
+            int prevDiscardActionCount = 0;
+    
+            // Now verify unprepares are needed.                 
+            for(int i = 0; i < iterations; ++i) {
+
+                // Verify current queue depth is expected.
+                assertSame(prevDiscardActionCount, con.getDiscardedServerPreparedStatementCount());
+                
+                try (SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement)con.prepareStatement(query)) {
+                    pstmt.execute(); // sp_executesql
+            
+                    pstmt.execute(); // sp_prepexec
+                    ++prevDiscardActionCount;
+
+                    pstmt.execute(); // sp_execute
+                }
+
+                // Verify clean-up is happening as expected.
+                if(prevDiscardActionCount > con.getServerPreparedStatementDiscardThreshold()) {
+                    prevDiscardActionCount = 0;
+                }
+
+                assertSame(prevDiscardActionCount, con.getDiscardedServerPreparedStatementCount());
+            }  
+
+            // Verify total cache use.
+            assertSame(iterations * 4, executeSQLReturnFirstInt(con, verifyTotalCacheUsesQuery));              
```

Comment:
Hi @TobiasSQL . We are finished with our internal testing and want to merge your PR. Could you please disable just this test for now? We still see failures in this test that needs investigation but as you said it is not a functional issue.  Thanks again :)